poetry/puzzle/provider.py

```diff
@@ -169,6 +169,7 @@ def search_for_vcs(self, dependency):  # type: (VCSDependency) -> List[Package]
             dependency.source,
             dependency.reference,
             name=dependency.name,
+            develop=dependency.develop,
```

Comment:
This is not the right place to put this, `get_package_from_vcs` should not have any notion of `develop`.

So, similarly to what is done in `search_for_directory`, we can do:

```python
package.develop = dependency.develop
```